src/js/adapters/fileReader.js

```diff
@@ -301,8 +310,11 @@ const readDataFromByte = (filePath, start, numberOfBytes) => {
 };
 
 const parseSelectedData = (data, start, numberOfBytes) => {
```

Comment:
I would wish to have parseSelectedData renamed to something that connects it more to the readDataFromByte workflow. Perhaps parseDataReadFromBytePositionInFile or parseDataFromBytePositionIntoLineArrayAndCreateReturnObject ?

Reply:
Hmm, while I agree that the name isn't the best at the moment (and I don't have a better suggestion), the suggested names feel too wordy and perhaps overly specific? In my opinion, a name should represent what a function does independently, and not in relation to other things around.   

The last suggestion might be on to something though, perhaps something like `parseByteDataIntoStringArray` is a good name, which describes what it does fairly accurately. Alternatively, in order to make it clear that metadata is also calculated, you could instead call it `parseByteDataIntoStringArrayWithMetadata`. Documentation of the return object might be done with comments, such as those documenting `parseLinesBackwards`.

Although, now that I'm writing all this out, I am realizing that this function should probably be split up to make things even clearer. One function shoudl be responsible for converting to a string array ( `parseByteDataIntoStringArray` ), and one should extract metdata ( `extractMetadataFromByteData` ), and perhaps single function to combine these into an object to return (I don't have good name for this one yet).

Reply:
Totally agree on the wordiness. The splitting into several functions sounds like a good idea.